src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -60,12 +84,12 @@ public TexlBinding BindBody(INameResolver nameResolver, IBinderGlue documentBind
             }
 
             bindingConfig = bindingConfig ?? new BindingConfig(this._isImperative);
-            var binding = TexlBinding.Run(documentBinderGlue, UdfBody, UserDefinitionsNameResolver.Merge(nameResolver, _parametersRecordType, functionNameResolver), bindingConfig, features: features);
+            _binding = TexlBinding.Run(documentBinderGlue, UdfBody, UserDefinitionsNameResolver.Create(nameResolver, _args, functionNameResolver), bindingConfig, features: features, rule: rule);
```

Comment:
Is there a reason we're doing this with a NameResolver, rather than treating the args as a ruleScope? Either way works, just curious.

Reply:
No reason, didn't know about the rule scope, happy to look into that, if that's a better way to go